transport-native-epoll/src/main/java/io/netty/channel/epoll/EpollEventLoop.java

```diff
@@ -45,15 +46,18 @@
  */
 class EpollEventLoop extends SingleThreadEventLoop {
     private static final InternalLogger logger = InternalLoggerFactory.getInstance(EpollEventLoop.class);
+    private static final AtomicLongFieldUpdater<EpollEventLoop> DEADLINE_UPDATER =
+            AtomicLongFieldUpdater.newUpdater(EpollEventLoop.class, "volatileDeadlineNanos");
 
     static {
         // Ensure JNI is initialized by the time this class is loaded by this time!
         // We use unix-common methods in this class which are backed by JNI methods.
         Epoll.ensureAvailability();
     }
 
-    // Pick a number that no task could have previously used.
-    private long prevDeadlineNanos = nanoTime() - 1;
+    private long prevDeadlineNanos = Long.MAX_VALUE;
+    private volatile long volatileDeadlineNanos = prevDeadlineNanos;
```

Comment:
As we not expect many `EpollEventLoop` instances we could just use an `AtomicLong` and so remove the updater :)